plugins/web/opentelemetry-plugin-react-load/README.md

```diff
@@ -0,0 +1,47 @@
+# OpenTelemetry Plugin React Load
+[![Gitter chat][gitter-image]][gitter-url]
+[![NPM Published Version][npm-img]][npm-url]
+[![Apache License][license-image]][license-image]
+
+This module provides *automated instrumentation for React lifecycles* for Web applications.
+
+## Installation
+
+```bash
+npm install --save @opentelemetry/plugin-react-load
+```
+
+## Usage
+
+```js
+import { BaseOpenTelemetryComponent } from '@opentelemetry/plugin-react-load';
+
+// Set once for the entire plugin
+BaseOpenTelemetryComponent.setLogger(logger);
+BaseOpenTelemetryComponent.setTracer('name', 'version');
+```
+
+To instrument components, extend `BaseOpenTelemetryComponent`:
```

Comment:
How can I instrument a functional component? Would I just wrap it as a HOC?

```tsx
const FuncComponent = ({...props}) => (<BaseOpenTelemetryComponent>
  <div {...props}>Hello World</div>
</BaseOpenTelemetryComponent>);
```

Reply:
Right now it doesn't support functional components, I was hoping this was okay because functional components don't have lifecycle methods. If we wanted to add tracing to it I was thinking that could be a separate strain of work to instrument `useEffect`. I was also thinking this would be okay since functional components are typically less complicated than class components so it would likely be more important for a user to want the class component to be instrumented, what do you think?

Reply:
Most of components that we write today are functional components since react hooks make class-based components unnecessary in most cases. As such, instrumentation of functional components will be a big plus.